src/MongoDB.Driver.Core/Core/ConnectionPools/ExclusiveConnectionPool.cs

```diff
@@ -394,17 +414,17 @@ private class AcquireConnectionHelper
             public AcquireConnectionHelper(ExclusiveConnectionPool pool)
             {
                 _pool = pool;
-            }
 
-            // public methods
-            public void CheckingOutConnection()
-            {
                 var handler = _pool._checkingOutConnectionEventHandler;
                 if (handler != null)
                 {
                     handler(new ConnectionPoolCheckingOutConnectionEvent(_pool._serverId, EventContext.OperationId));
                 }
```

Comment:
Let's NOT raise this event from the constructor. Let's leave it in `CheckingOutConnection` like it was.

Reply:
done